### DISCUSSION TITLE
Power1 1, PWM Signal whilst button pressed

### DISCUSSION THREAD

**Discussion by sag007** [arendst/Tasmota#22469] [Q&A]
Hey Guys,

I have a quick one, I have setoption73 1 to detatch the relays from the buttons because I'm trying to drive a shutter that requires a PMW of around 250 to close and around the 450 to open. Also at this stage there are no limits installed so I only want to drive the shutter whilst the buttons are pressed.

Ideally to open the shutter I need to use button1 only whilst pressed to set power1 1 and set pwm1 to 250 when pressed and set power1 0 and pwm1 to 300 when released.
On the flip side, to close the shutter I need button2 only whilst pressed to set power1 1 and set pwm450, then when released set power1 0 and set pwm1 300.

I would be grateful for any suggestions you could offer as I haven't done anything like this one before.

Regards,

Nathan

**Comment by sfromis** [comment]
To handle "whilst pressed", you could reconfigure the Button as a Switch instead (with `SetOption114`), and use something like `SwitchMode2 1`, follow mode, which will emit different states for pushed and released. Those you could pick up with rules to issue your `Power` and `PWMFrequency` commands.

**Reply by sag007** [reply]
So I have both the buttons as switches and switchmode set to 2 on both buttons with the following rules; 

rule1 on switch1#state do if (switch1#state==0) backlog power1 1;  pwm1 250 elseif (switch1#state==1) backlog power1 0; pwm1 300 endif endon
rule2 on switch2#state do if (switch2#state==0) backlog power1 1;  pwm1 450 elseif (switch2#state==1) backlog power1 0; pwm1 300 endif endon

I also tried to set both switches to switchmode 7 but still no good.

**Reply by sfromis** [reply]
It may help with valid rules instead of *that*....
You cannot use trigger expressions with `#` anywhere but in the trigger part before `do`
In your case, you could have 4 rules with `on Switch2#state==1 do ... endon` etc.
Or with `if`, you *could* also use `%value%` to refer to the value from `on`.

**Comment by sag007** [comment]
Hey stfromis,

Yeah ended up with the following rules that seem to work pretty well;
rule1 on switch1#state==1 do backlog power1 1; pwm3 400 endon on switch1#state==0 do backlog power1 0; pwm3 300 endon
rule2 on switch2#state==1 do backlog power1 1; pwm3 250 endon on switch2#state==0 do backlog power1 0; pwm3 300 endon

Cheers.

**Reply by sag007** [reply]
Hey Guys,
I would like to enable shutter on this cover now and add the buttons to it but think I'm going to run into an issue with this servo that I have had to use.
This servo is a 12v unit and is also 360 degree continuous rotation. With this servo the pwm range is 155 to 490, however the stop/dead zone is actually a pwm of 300. I have enabled shuttermode 6 with the following settings but this fails;
SetOption80 1
ShutterRelay1 1
ShutterMode 6
PWMfrequency 200
SetOption15 0
ShutterPwmRange1 155 490
ShutterOpenDuration1 10.0
ShutterCloseDuration1 10.0
ShutterMotorDelay1 2.0

My thoughts are to ultimately have at least one proximity switch at one end so I can reset the cover each time it is closed and then drive for a set time period when opening. Also to be able to control it with local hard wired buttons or soft buttons through HA.

Any suggestions would be greatly appreciated, even to know this isn't possible helps.
